File: Exam/src/main/java/com/example/exam/service/impl/ToolService.java
package com.example.exam.service.impl;

import com.example.exam.model.Tool;
import com.example.exam.repository.IToolRepository;
import com.example.exam.service.IToolService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public class ToolService implements IToolService {
    @Autowired
    private IToolRepository iToolRepository;
    @Override
    public Page<Tool> findByNameTool(String name, Pageable pageable) {
        return iToolRepository.findByNameTool(name,pageable);
    }

    @Override
    public void toolCreate(Tool tool) {
        iToolRepository.save(tool);
    }

    @Override
    public Boolean existsByCode(String code) {
        return iToolRepository.existsByCode(code);
    }
}

File: examfe/src/service/ToolService.js
import axios from 'axios'
export const findByNameTool = (name,page)=>{
    try {
        const rs = axios.get(`http://localhost:8080/tool?name=${name?name:''}&page=${page?page:0}`)
        return rs
    } catch (error) {
        console.log(error);
    }
}
export const toolCreate = (value)=>{
    try {
        const rs = axios.post(`http://localhost:8080/tool`,{ ...value })
        return rs
    } catch (error) {
        console.log(error);
    }
}
